lsp-protocol.el

```diff
@@ -475,6 +475,7 @@ See `-let' for a description of the destructuring mechanism."
  (TextDocumentItem (:languageId :text :uri :version) nil)
  (TextDocumentSyncOptions nil (:change :openClose :save :willSave :willSaveWaitUntil))
  (TextEdit (:newText :range) nil)
+ (SnippetTextEdit (:newText :range) (:insertTextFormat))
```

Comment:
is this part of LSP spec? or is rust specific? if so should this go [here](https://github.com/emacs-lsp/lsp-mode/pull/1926/files#diff-459ac98aed37663772c354a97a7006a1R270)?

Reply:
it is implemented in vscode experimental features, at least I was unable to find it in rust analyzer client. My guess is that it will make to the spec at some point.